ocaml/testsuite/tests/typing-layouts/basics_alpha.ml

```diff
@@ -1738,7 +1741,7 @@ Error: This expression has type t_void but an expression was expected of type
        The layout of t_void is void, because
          of the annotation on the declaration of the type t_void.
        But the layout of t_void must be a sublayout of value, because
-         it's used as a function argument.
+         it's used as a function argument, defaulted to layout value.
```

Comment:
I'm worried that this message might be wrong in a closely related scenario. Try changing the sig for `f` to be `val f : 'a -> ('a : value)`. Does the output still say "defaulted"? If it doesn't, then yes let's merge.